geocoder_api/routers/geocoding.py

```diff
@@ -1,17 +1,22 @@
 from fastapi import APIRouter, HTTPException, Query
+from fastapi_cache.decorator import cache
 from typing_extensions import Annotated
 from geocoder_api.settings import Settings
 from httpx import AsyncClient
+from geojson_pydantic import FeatureCollection
 
 settings = Settings()
-router = APIRouter(prefix="/geocoding", tags=["geocoding"])
+router = APIRouter(tags=["geocoding"])
 
 
 @router.get(
     "/",
     description="Returns a GeoJSON FeatureCollection of places matching the search query",
 )
-async def geocode(q: Annotated[str, Query(description="Search query")]):
+@cache(expire=3600)  # Expires after 1 hour
+async def geocode(
+    q: Annotated[str, Query(description="Search query")]
+) -> FeatureCollection:
```

Comment:
This FeatureCollection does not provide any specification about what type of properties the returned geojson-features have?

Is it possible to also model them, so we can have a full documentation of the returned data?